src/main/java/edu/bu/metcs673/trackr/service/impl/BankAccountServiceImpl.java

```diff
@@ -97,4 +100,17 @@ public BankAccount findBankAccountByIdAndUserId(long id, long userId) {
 
     }
 
+    @Override
+    public BankAccount findBankAccountByUserId(long userId) {
```

Comment:
The implementation is not good since it needs to use both the account id  you are looking for and the user id.